.github/workflows/action.yml

```diff
@@ -51,6 +51,15 @@ jobs:
         working-directory: ${{env.path}}/build/unit_tests
         run: make -j 3
 
+  generate-docs:
+    runs-on: self-hosted
+    needs: checkout-and-setup
+    if: github.ref == 'refs/heads/main'
```

Comment:
why do we only generate docs on main?

Reply:
Because we only want docs for main.

Reply:
Imo we should generate docs for all branches, that way docs can be reviewed before they are merged.

Reply:
sure, but where would they go?

Reply:
Decided that setting this up will be time expensive and of limited benefit when the documentation can still be generated locally for branches